scripts/daqconf_viewer

```diff
@@ -0,0 +1,261 @@
+            self.confdata = r.json()
+
+    async def watch_confdata(self, confdata:dict) -> None:
+        '''Turns the jsons into a string format with newlines, then generates a diff of the two.'''
+        if confdata:
+            j1 = copy.deepcopy(oldconf)
+            j2 = copy.deepcopy(confdata)
+            if "_id" in j1: del j1["_id"]       #We don't want to include the ID in the diff since it's always different.
+            if "_id" in j2: del j2["_id"]
+            a = json.dumps(j1, sort_keys=True, indent=4).splitlines(keepends=True)
+            b = json.dumps(j2, sort_keys=True, indent=4).splitlines(keepends=True)
+            delta = unified_diff(a,b)
+            diff = Text()
+            for d in delta:
+                sym = d[0]
+                match sym:
+                    case '+':
+                        t = Text(d, style='green')
+                    case '-':
+                        t = Text(d, style='red')
+                    case '@':
+                        t = Text(d, style='gold1')
+                    case _:
+                        t = Text(d)
+                diff += t
+
+            box = self.query_one(Static)
+            box.update(diff)
+
+    def on_button_pressed(self) -> None:
+        self.remove()
+
+class DiffScreen(Screen):
+    BINDINGS = [("d", "app.pop_screen", "Return")]
+    def __init__(self, hostname, **kwargs):
+        super().__init__(**kwargs)
+        self.hostname = hostname
+
+    def compose(self) -> ComposeResult:
+        yield Configs(hostname=self.hostname, classes='greencontainer configs', id='diffconfigs')
+        yield Versions(hostname=self.hostname, classes='greencontainer versions', id='diffversions')
+        yield DiffDisplay(hostname=self.hostname, classes='greencontainer display', id='diffdisplay')
+
+        yield Header(show_clock=True)
+        yield Footer()
+
+
+class ConfViewer(App):
+    CSS_PATH = "daqconf_viewer.css"
+    BINDINGS = [("d", "make_diff()", "Diff")]
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.hostname = "http://np04-srv-023:31011"
+
+    def on_mount(self) -> None:
+        self.install_screen(DiffScreen(hostname=self.hostname), name="diff")
+
+    def compose(self) -> ComposeResult:
+        yield Configs(hostname=self.hostname, classes='redcontainer configs', id='regconfigs')
+        yield Versions(hostname=self.hostname, classes='redcontainer versions', id='regversions')
+        yield Display(hostname=self.hostname, classes='redcontainer display', id='regdisplay')
+
+        yield Header(show_clock=True)
+        yield Footer()
+
+    def action_make_diff(self):
+        '''Saves the current config to a global variable, then pushes the diff screen.'''
+        dis = self.query_one(Display)
+        if dis.confdata != None:
+           global oldconf
+           oldconf = dis.confdata
+           self.push_screen('diff')
+
+if __name__ == "__main__":
```

Comment:
Could you import click and make a `--help` function here? We also may want to use a different hostname (the one in line 238) for the mongo db, it would be nice to be able to specify it in the CLI.